torch/serialization.py

```diff
@@ -1105,6 +1105,21 @@ def load(
             )
     else:
         if pickle_module is None:
+            if weights_only is None:
+                warnings.warn(
+                    "You are using `torch.load` with `weights_only=False` (the current default value), which uses "
+                    "the default pickle module implicitly. It is possible to construct malicious pickle data "
+                    "which will execute arbitrary code during unpickling (See "
+                    "https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models for more details). "
+                    "In a future release, the default value for `weights_only` will be flipped to `True`. This "
+                    "limits the functions that could be executed during unpickling. Arbitrary objects will no "
+                    "longer be allowed to be loaded via this mode unless they are explicitly allowlisted by the "
+                    "user via `torch.serialization.add_safe_globals`. We recommend you start setting "
+                    "`weights_only=True` for any use case where you don't have full control of the loaded file. "
+                    "Please open an issue on GitHub for any issues related to this experimental feature.",
+                    FutureWarning,
+                )
+                weights_only = False
```

Comment:
nit: does the code need updating to handle None vs False? I guess the code below "just works" because they both behave the same when you do "if weights_only". But we should be consistent and only set it to False here if we always do so. Otherwise never change it.

Reply:
ok, fixed this to always set to False up top